### PR TITLE
Support for new matrix crate

### LINKED ISSUE #12: Integration with other matrix librairies
It should be possible to create an example with other matrix librairies.

Below is a not an exhaustive list but it can be a good target for this first issue: 

- [x]  [rulinalg](https://crates.io/crates/rulinalg) 
- [x]  [nalgebra](https://crates.io/crates/nalgebra)

### PR DESCRIPTION
**Pull request by ThomAub**
This is an inital PR to fix #12

### CODE REVIEW

**Comment by ThomAub** [comment]
I will fix it ! 
It seems to be a clippy problem

File: Cargo.toml [hunk 1]
```diff
@@ -19,11 +19,13 @@ showata = {version = "0.2.0", optional = true}
...
+nalgebra = {version ="0.19.0", features = ["serde-serialize"], optional=true}
 
 [dev-dependencies]
 csv = "1.1"
 ndarray-rand = "0.11"
 
 [features]
 show_vega = ["showata", "anyhow"]
-default = ["show_vega", "csv", "ndarray"]
+default = ["show_vega", "csv", "ndarray", "nalgebra", "rulinalg"]
```
Comment (mockersf) [inline Cargo.toml]: I'm not sure we want to have `ndarray`, `nalgebra` and `rulinalg` by default. We should either pick one or make all three opt-ins

Comment (mockersf) [reply Cargo.toml]: Also, features should be explained in project `README.md`

Comment (ThomAub) [reply Cargo.toml]: I agree. Currently only `ndarray` and `nalgebra` are in active development so I think we can make it one of it as default or all three as opt-ins.
I will push a updated project `README.md` the one in example is already updated.

Comment (ThomAub) [reply Cargo.toml]: Fixed by daadf47 :)

**Review by davidB — CHANGES_REQUESTED** [review]
instead of having a single data.rs file with group of cfg macro, I prefer to have the module as folder with 1 file per cfg and only use the cfg macro when the mod is imported in mod.rs

**Review by davidB — APPROVED** [review]

**Comment by ThomAub** [comment]
I don't if it's the best way to do it but i hope it does what you asked @davidB ?

File: examples/README.md [hunk 2]
```diff
@@ -130,6 +133,26 @@ Create a chart from existing json and add new data
 
 <img src="https://raw.githubusercontent.com/procyon-rs/vega_lite_3.rs/master/examples/res/screens/mixed.png">
 
+## [From nalgebra](https://github.com/procyon-rs/vega_lite_3.rs/blob/master/examples/from_nalgebra.rs)
```
Comment (mockersf) [inline examples/README.md]: not the good level of header, should be `###`

File: examples/README.md [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
-  - [Line with confidence Interval](#line-with-confidence-interval)
-- [Data Sources](#data-sources)
-  - [From ndarray](#from-ndarray)
-  - [From csv](#from-csv)
-  - [From url](#from-url)
-  - [From JSON spec](#from-json-spec)
-  - [From JSON spec with new Data](#from-json-spec-with-new-data)
-- [Miscellaneous](#miscellaneous)
-  - [Without using builders](#without-using-builders)
+- [Examples Gallery](#examples-gallery)
```
Comment (mockersf) [inline examples/README.md]: when generating the TOC, don't add the level 1 header as it's just above

File: examples/README.md [hunk 4]
```diff
@@ -130,6 +133,26 @@ Create a chart from existing json and add new data
...
+
+```rust
+cargo run --example from_nalgebra --features nalgebra
+```
+
+Loads data directly from a random [`nalgebra::Matrix`](https://docs.rs/nalgebra/0.19.0/nalgebra/index.html).
+
+<img src="https://raw.githubusercontent.com/procyon-rs/vega_lite_3.rs/master/examples/res/screens/rulinalg_nalgebra.png">
+
+## [From rulinalg](https://github.com/procyon-rs/vega_lite_3.rs/blob/master/examples/from_ndarray.rs)
```
Comment (mockersf) [inline examples/README.md]: should be `###`

**Review by mockersf — CHANGES_REQUESTED** [review]

**Review by mockersf — APPROVED** [review]

**Pull request merged**